bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java

```diff
@@ -505,6 +539,22 @@ public List<Long> getReceiverChatIds() {
 
     public void addMessageId(Long chatId, String replyId, Integer messageId) {
         replyIdToMessageId.put(new ReplyKey(chatId, replyId), messageId);
+
+        // persist current replyIdToMessageId entries in thing configuration
+        JsonObject root = new JsonObject();
+        for (Map.Entry<ReplyKey, Integer> entry : replyIdToMessageId.entrySet()) {
+            @Nullable
+            ReplyKey replyKey = entry.getKey(); // dein ReplyKey.toString()
+            @Nullable
+            Integer intsgId = entry.getValue();
+            String strMsgId = intsgId.toString();
+            String strValue = replyKey.getChatId() + "|" + replyKey.getReplyId();
+            root.add(strMsgId, new JsonPrimitive(strValue));
+        }
+        String jsonString = root.toString();
+        Configuration cfg = editConfiguration();
+        cfg.put("replyIdToMessageIdMappings", jsonString);
+        updateConfiguration(cfg);
```

Comment:
I don't think this is the proper way to store data from a binding. You should use a `Storage` for that. Inject the `StorageService` into the thing handler factory and create a `Storage`. Pass that storage to the thing handler and use the `ThingUID` as key. You can store arbitrary data in the `Storage`, so no need to serialize/de-serialize the information manually.